### PR TITLE
chore: deprecate latestAmazonLinux2022()

### LINKED ISSUE #26274: ec2: latestAmazonLinux2022() does not return available parameters
### Describe the bug

`ec2.MachineImage.latestAmazonLinux2022()` does not return any valid Amazon Linux 2022 parameters.

Looking at its implementation, it's supposed to return parameters with `al2022` prefix

https://github.com/aws/aws-cdk/blob/7f31d9fc130002e8010a47d432b1277d76b9d835/packages/aws-cdk-lib/aws-ec2/lib/machine-image/amazon-linux-2022.ts#L54-L68

But if  we list all parameters with `/aws/service/ami-amazon-linux-latest` path, no `al2022` parameters will be returned.

```
$ aws ssm get-parameters-by-path --path /aws/service/ami-amazon-linux-latest --query "Parameters[].Name"
[
    "/aws/service/ami-amazon-linux-latest/al2023-ami-kernel-6.1-arm64",
    "/aws/service/ami-amazon-linux-latest/al2023-ami-kernel-6.1-x86_64",
    "/aws/service/ami-amazon-linux-latest/al2023-ami-minimal-kernel-6.1-arm64",
    "/aws/service/ami-amazon-linux-latest/al2023-ami-minimal-kernel-6.1-x86_64",
    "/aws/service/ami-amazon-linux-latest/al2023-ami-minimal-kernel-default-arm64",
    "/aws/service/ami-amazon-linux-latest/amzn-ami-hvm-x86_64-gp2",
    "/aws/service/ami-amazon-linux-latest/amzn-ami-hvm-x86_64-s3",
    "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-ebs",
    "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2",
    "/aws/service/ami-amazon-linux-latest/amzn2-ami-kernel-5.10-hvm-x86_64-ebs",
    "/aws/service/ami-amazon-linux-latest/al2023-ami-kernel-default-x86_64",
    "/aws/service/ami-amazon-linux-latest/al2023-ami-minimal-kernel-default-x86_64",
    "/aws/service/ami-amazon-linux-latest/amzn-ami-hvm-x86_64-ebs",
    "/aws/service/ami-amazon-linux-latest/amzn-ami-minimal-hvm-x86_64-s3",
    "/aws/service/ami-amazon-linux-latest/amzn-ami-minimal-pv-x86_64-s3",
    "/aws/service/ami-amazon-linux-latest/amzn-ami-pv-x86_64-s3",
    "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-arm64-gp2",
    "/aws/service/ami-amazon-linux-latest/amzn2-ami-kernel-5.10-hvm-arm64-gp2",
    "/aws/service/ami-amazon-linux-latest/amzn2-ami-kernel-5.10-hvm-x86_64-gp2",
    "/aws/service/ami-amazon-linux-latest/amzn2-ami-minimal-hvm-arm64-ebs",
    "/aws/service/ami-amazon-linux-latest/al2023-ami-kernel-default-arm64",
    "/aws/service/ami-amazon-linux-latest/amzn-ami-minimal-hvm-x86_64-ebs",
    "/aws/service/ami-amazon-linux-latest/amzn-ami-minimal-pv-x86_64-ebs",
    "/aws/service/ami-amazon-linux-latest/amzn-ami-pv-x86_64-ebs",
    "/aws/service/ami-amazon-linux-latest/amzn2-ami-minimal-hvm-x86_64-ebs"
]
```

I guess we probably should deprecate `latestAmazonLinux2022()`.

### Expected Behavior

parameters returned from latestAmazonLinux2022() should be available.

### Current Behavior

parameters not found.

### Reproduction Steps

ec2.MachineImage.latestAmazonLinux2022()

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.87.0

### Framework Version

_No response_

### Node.js Version

v18.16.0

### OS

mac os x

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by pahud**
SSM parameter is not offering any al2022 AMIs as described in https://github.com/aws/aws-cdk/issues/26274. This PR marks `latestAmazonLinux2022()` as deprecated and uses `latestAmazonLinux2023()` instead. 

- [x] mark latestAmazonLinux2022 as deprecated
- [x] update the aws-ec2 README to use latestAmazonLinux2023 instead

Closes #26274 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ec2/README.md [hunk 1]
```diff
@@ -1070,18 +1070,18 @@ new ec2.Instance(this, 'Instance3', {
   }),
 });
 
-// AWS Linux 2022
+// AWS Linux 2023
```
Comment (markusl) [inline packages/aws-cdk-lib/aws-ec2/README.md:1073]: Amazon?

Comment (mrgrain) [reply packages/aws-cdk-lib/aws-ec2/README.md:1073]: Yes, should be 
```suggestion
// Amazon Linux 2023
```

**Comment by mrgrain** [comment]
@pahud can we find out if these images are still published and hardcode the latest values so people are not completely broken when using `latestAmazonLinux2022()`?

**Comment by pahud** [comment]
> @pahud can we find out if these images are still published and hardcode the latest values so people are not completely broken when using `latestAmazonLinux2022()`?

I can confirm `latestAmazonLinux2022()` does not resolve any Ami from the generated parameter path and the user will receive this error when `cdk deploy`:

```
 test-ec2-stack failed: Error [ValidationError]: Unable to fetch parameters [/aws/service/ami-amazon-linux-latest/al2022-ami-kernel-5.15-x86_64] from parameter store for this account.
```

This means:

1. Users previously successfully deploy with `latestAmazonLinux2022()` will not be able to redeploy because cloudformation will not find the original parameter path and AFAIK there's no amazon linux 2022 latest AMI anymore.
2. They will have to choose other MachineImage and update their stack either using the original AmiId or `latestAmazonLinux2023()`

I am not sure if we should completely remove the `latestAmazonLinux2022()` method but this PR just adds the deprecated tag on it.

**Comment by mrgrain** [comment]
@pahud I meant can we find out what the actual AMI image ID is and put that in?

**Review by mrgrain — APPROVED** [review]

**Review by mrgrain — APPROVED** [review]

**Pull request merged**